### ISSUE TITLE
preserving comments

### ISSUE THREAD

**Issue by wmtorode** [dpranke/pyjson5#28]
Is there a way to preserve comments so that if I edit a file the comments it contained are still there?

**Comment by dpranke** [comment]
Unfortunately, no, not currently. 

In order to do that you need to preserve the whole parse tree, which certainly doesn't work inside the existing simple load/dump API.

I would like to support this, though, it's something of a challenging problem. I've never seen an API for doing something like that and so it's a bit unclear what a good API would look like.

**Comment by wmtorode** [comment]
Thanks, I figured that would be the answer. It would be great if this could be supported even if it's through a different API than the current load and dump.

**Comment by dpranke** [comment]
Agreed. (Sorry for not responding sooner. I'm still mulling over what an API like this might look like.)

**Comment by qneill** [comment]
This npm package is doing it (or claiming to): https://www.npmjs.com/package/comment-json

**Comment by dpranke** [comment]
@qneill Thanks for the pointer. I'm looking at that now, and it does seem to be at least reasonable well-worked out.

**Comment by dpranke** [comment]
Thanks for the links! Wow, has it really been 5 years since I've commented on this? That's kinda sad ...

I was actually working on this earlier this year and made a good deal of progress before I got distracted by something else. I'll take a look at the things you're pointing at (and comment-json again) and see how they compare to what I had in mind.

Tangentially, unless you had a specific reason to use JSON5 for a project (like existing JSON5 files, or something strictly compatible with JavaScript), I probably wouldn't. There are other similar formats that are more popular and more powerful. YAML is the obvious frontrunner, but if you wanted something a lot lot simpler, I would probably point you at [HJSON](https://hjson.github.io/). I actually have a different format that is very similar to HJSON but has some things I consider advantages, but it's not particular public or suited for consumption by others at this point, so I won't point you at it.
